### PR TITLE
Re-write the dropzone component and fix styling issues

### PR DESCRIPTION
**Pull request by farhaanbukhsh**
## Description

We are planning to use the Dropzone Paragone component here and also fix the style for URL video uploader. We are trying to keep the paragon component intact for it to be used.

Introduced new thumbnail

image

Introduced URL uploader field

image

We remove the custom Dropzone that was written initially and have used the paragon inbuilt one.

## Testing instructions

1. Set up the master devstack, make sure you have course-authoring-mfe(frontend-app-course-authoring) running
2. We have to clone this repo in {EDX_DEVSTACK}/src directory.
3. Change the branch being used to `open-craft:farhaan/fix-drag-drop-component` and run `make build`
4. Follow the steps mentioned in https://github.com/openedx/frontend-lib-content-components/pull/262 to set up video editor, make sure to `Step 3 - Step 7`
5. We need to upload a small video and connect it to the demo course as mentioned in https://github.com/openedx/frontend-lib-content-components/pull/326
6. We need to make changes to the edx-platform to dummy S3 connection as mentioned in the above PR.
7. Now add a video xblock in any unit, you will be redirected to the the course-authoring-mfe's video setting page.
8. Check the thumbnail here -- new addition
8. Go to "Replace Video"
9. Try to upload a video, you should see a spinner here.
10. You will see the new URL bar here as well -- this works like a search field.

## Other information

Private ref: https://tasks.opencraft.com/browse/BB-7524

Reviewer:

- [ ] @0x29a

### CODE REVIEW

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 1]
```diff
@@ -11,77 +12,69 @@ import * as hooks from './hooks';
...
-  const { textInputValue, setTextInputValue } = hooks.uploader();
-  const onURLUpload = hooks.onVideoUpload();
+const URLUploader = () => (
+  <div className="d-flex flex-column flex-wrap">
+    <div className="justify-content-center align-self-center bg-light rounded-circle p-4">
+      <Icon src={FileUpload} className="text-muted" />
+    </div>
+    <div className="d-flex align-self-center justify-content-center flex-wrap flex-column pt-5">
+      <span style={{ fontSize: '20px' }}><FormattedMessage {...messages.dropVideoFileHere} /></span>
+      <span className="align-self-center" style={{ fontSize: '12px' }}><FormattedMessage {...messages.info} /></span>
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.jsx:22]: I think using FormattedMessage like this doesn't provide any real benefit over just using `intl.formatMessage`, so I'd recommend that instead.

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 2]
```diff
@@ -11,77 +12,69 @@ import * as hooks from './hooks';
...
-    multiple: false,
-    onDrop: (acceptedFiles) => {
-      if (acceptedFiles.length > 0) {
-        const uploadfile = acceptedFiles[0];
-        onUpload(uploadfile);
-      }
-    },
-  });
+export const VideoUploader = ({ onUpload }) => {
+  const { textInputValue, settextInputValue } = hooks.uploader();
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.jsx:29]:
```suggestion
  const { textInputValue, setTextInputValue } = hooks.uploader();
```

Comment (xitij2000) [reply src/editors/containers/VideoUploadEditor/index.jsx:29]: I am not seeing any benefit in calling hooks.uploader, which then calls testInputValue when then calls useState. 

Just directly have React.useState here.

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 3]
```diff
@@ -120,7 +107,7 @@ const VideoUploadEditor = (
               onClick={handleCancel}
             />
           </div>
-          <VideoUploader onUpload={handleDrop} errorMessage={errorMessage} intl={intl} />
+          <VideoUploader onUpload={handleDrop} intl={intl} />
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.jsx:110]: Why are we passing `intl` here? We can just use `useIntl` in that component.

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 4]
```diff
@@ -11,77 +12,69 @@ import * as hooks from './hooks';
...
-    onDrop: (acceptedFiles) => {
-      if (acceptedFiles.length > 0) {
-        const uploadfile = acceptedFiles[0];
-        onUpload(uploadfile);
-      }
-    },
-  });
+export const VideoUploader = ({ onUpload }) => {
+  const { textInputValue, settextInputValue } = hooks.uploader();
+  const onURLUpload = hooks.onVideoUpload();
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.jsx:30]: This is about onVideoUpload. 

That function is doing things it shouldn't. 

1. Instead of importing store directly you can use the `useStore` hook. 
2. If you want to pass the store as a function argument, just call that function with `useSelector` i.e. `useSelector(postUploadRedirect`. 

I think what you should do instead is, call the selector `postUploadRedirectUrl` as follows:

```
const postUploadRedirectUrl = createSelector([learningContextId, blockId], (contextId, blockId) =>`/course/${learningContextId}/editor/video/${blockId}?selectedVideoUrl=${videoUrl}`). 
```
Then use it as:
```
const redirectUrl = useSelector(postUploadRedirectUrl);
...
// In some component:
//onClick={() => navigateTo(redirectUrl)
```

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 5]
```diff
@@ -11,77 +12,69 @@ import * as hooks from './hooks';
...
+  const URLUploaderComponent = URLUploader();
 
-  const handleInputChange = (event) => {
-    setTextInputValue(event.target.value);
+  const handleInputChange = (value) => {
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.jsx:33]: This function isn't doing anything, just set onChange to setTextInputChange directly.

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 6]
```diff
@@ -11,77 +12,69 @@ import * as hooks from './hooks';
...
-    );
-  }
+  const handleProcessUpload = async ({
+    // eslint-disable-next-line no-unused-vars
+    fileData, requestConfig, handleError,
+  }) => {
+    try {
+      await onUpload(fileData);
+    } catch (error) {
+      console.log(error);
```
[the remark this answers is not in the document]
Comment (xitij2000) [reply src/editors/containers/VideoUploadEditor/index.jsx:48]: Currently the errors are being swallowed. I think there should be some response to users.

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 7]
```diff
@@ -11,77 +12,69 @@ import * as hooks from './hooks';
...
   };
 
-  if (errorMessage) {
-    return (
-      <div className="d-flex flex-column justify-content-center align-items-center text-center error-message">{errorMessage}</div>
-    );
-  }
+  const handleProcessUpload = async ({
+    // eslint-disable-next-line no-unused-vars
+    fileData, requestConfig, handleError,
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.jsx:43]: If the params are not used just remove them?

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 8]
```diff
@@ -11,77 +12,69 @@ import * as hooks from './hooks';
...
-          </button>
-        </div>
+      <Dropzone
+        accept={{ 'video/*': ['.mp4', '.mov'] }}
+        onProcessUpload={handleProcessUpload}
+        inputComponent={URLUploaderComponent}
+      />
+      <div className="d-flex video-id-prompt">
+        <SearchField
+          placeholder="Paste your video ID or URL"
```
[the remark this answers is not in the document]
Comment (xitij2000) [reply src/editors/containers/VideoUploadEditor/index.jsx:61]: Additionally, why is this a search field?

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 9]
```diff
@@ -95,17 +88,11 @@ const VideoUploadEditor = (
   const {
     loading,
     setLoading,
-    errorMessage,
-    setErrorMessage,
   } = hooks.uploadEditor();
   const handleCancel = editorHooks.handleCancel({ onClose });
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.jsx:92]: It seems if you provide onClose as an argument to this function it returns that function, so it is doing nothing here. This function take onClose and returns onClose. So this call can be skipped.

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 10]
```diff
@@ -95,17 +88,11 @@ const VideoUploadEditor = (
   const {
     loading,
     setLoading,
-    errorMessage,
-    setErrorMessage,
   } = hooks.uploadEditor();
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.jsx:91]: This can also be simplified to a React.useState call.

File: src/editors/containers/VideoUploadEditor/hooks.js [hunk 11]
```diff
@@ -54,22 +49,17 @@ export const onVideoUpload = () => {
...
-  } else {
-    const errorMsg = 'Video must be an MP4 or MOV file';
-    setErrorMessage(errorMsg);
-  }
+  uploadVideo({
+    supportedFiles: [newFile],
+    setLoadSpinner: setLoading,
+    postUploadRedirect: onVideoUpload(),
+  });
 };
```
[the remark this answers is not in the document]
Comment (xitij2000) [reply src/editors/containers/VideoUploadEditor/hooks.js:63]: @0x29a I think the dropzone component already does validation, so it might be OK to skip it here. 

That said, I don't understand what this function is doing. Are browsers not filling in the mime type? I tried a couple of files and the type is already set correctly, so you can just use the original file instead of creating a new one. I agree that it's no longer a validator, so the name should change. I think removing that function can calling uploadVideo directly would be better.

**Review by xitij2000 — CHANGES_REQUESTED** [review]

File: src/editors/containers/VideoUploadEditor/index.scss [hunk 12]
```diff
@@ -4,11 +4,15 @@
...
   }
+}
 
-  .file-upload {
-    width: 56px;
-    height: 56px;
-  }
+.dropzone-file-container {
+  height: 100vh;
+}
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.scss:11]: I don't see this being used.

File: src/editors/containers/VideoUploadEditor/index.scss [hunk 13]
```diff
@@ -32,12 +34,22 @@
...
     border: none !important;
     background-color: #FFFFFF;
   }
+
+  .prompt-button {
+    background: rgba(239, 234, 247, 0.70);
+  }
 }
 
 .error-message {
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.scss:55]: A lot of this file contains magic values. Can you instead use variables so that it picks up the correct theme?

You can see the variables here: https://paragon-openedx.netlify.app/foundations/css-utilities/
Find the matching variable for what you want. I think you might need to look at the edX theme, since I think figma designs generally use the edX theme.

**Review by xitij2000 — CHANGES_REQUESTED** [review]

File: src/editors/containers/VideoUploadEditor/hooks.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (farhaanbukhsh) [reply src/editors/containers/VideoUploadEditor/hooks.js:63]: @0x29a @xitij2000  you are write we no longer need this function since Dropzone can handle this.

File: src/editors/containers/VideoUploadEditor/index.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (farhaanbukhsh) [reply src/editors/containers/VideoUploadEditor/index.jsx:48]: Probably but we are not making it reach the user here. May be in the next iteration.

File: src/editors/containers/VideoUploadEditor/index.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (farhaanbukhsh) [reply src/editors/containers/VideoUploadEditor/index.jsx:61]: I have changed this to a inputGroup now

File: src/editors/containers/VideoUploadEditor/index.scss (hunk 13 above)
Comment (farhaanbukhsh) [reply src/editors/containers/VideoUploadEditor/index.scss:55]: I have tried removing as much as I can :)

File: src/editors/containers/VideoUploadEditor/index.jsx (hunk 9 above)
Comment (farhaanbukhsh) [reply src/editors/containers/VideoUploadEditor/index.jsx:92]: It is going with the editor component which this component is wrapped in, FWIW it's telling the parent component about it. I am keeping this for now.

File: src/editors/containers/VideoUploadEditor/index.jsx (hunk 3 above)
Comment (farhaanbukhsh) [reply src/editors/containers/VideoUploadEditor/index.jsx:110]: useIntl is not available to us as pointed out in https://github.com/openedx/frontend-lib-content-components/blob/main/docs/decisions/0005-internal-editor-testability-decisions.md#additional-considrations

Comment (xitij2000) [reply src/editors/containers/VideoUploadEditor/index.jsx]: I think it's available now, I've used it in my latest PR.

File: src/editors/containers/VideoUploadEditor/index.jsx (hunk 9 above)
Comment (xitij2000) [reply src/editors/containers/VideoUploadEditor/index.jsx:96]: I'm not sure I understand what you mean? `handleCancel` is just a regular function. I think you can replace this with
`  const handleCancel = onClose;` and it will work the same. 

It's not part of this PR, so it's OK to leave it for now.

File: src/editors/containers/VideoUploadEditor/index.jsx [hunk 14]
```diff
@@ -11,77 +12,73 @@ import * as hooks from './hooks';
 import messages from './messages';
 import * as editorHooks from '../EditorContainer/hooks';
 
-export const VideoUploader = ({ onUpload, errorMessage }) => {
-  const { textInputValue, setTextInputValue } = hooks.uploader();
-  const onURLUpload = hooks.onVideoUpload();
+const URLUploader = ({ intl }) => (
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.jsx:15]: You can use `const intl = useIntl()` here. I think the advice against that is outdated since this uses a newer version of platform now which includes that.

File: src/editors/containers/VideoUploadEditor/hooks.js [hunk 15]
```diff
@@ -52,30 +18,21 @@ export const onVideoUpload = () => {
...
-    uploadVideo({
-      supportedFiles: [newFile],
-      setLoadSpinner: setLoading,
-      postUploadRedirect: onVideoUpload(),
-    });
-  } else {
-    const errorMsg = 'Video must be an MP4 or MOV file';
-    setErrorMessage(errorMsg);
-  }
+export const useUploadVideo = async ({
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/hooks.js:21]: This is not a react hook, and so should not be named starting with use. 
Additionally, its not adding much so perhaps the dispatch call can be moved directly into the component?
Is the idea to wrap it here to make it easier to mock?

**Review by xitij2000 — CHANGES_REQUESTED** [review]

File: src/editors/containers/VideoUploadEditor/index.jsx (hunk 9 above)
Comment (farhaanbukhsh) [reply src/editors/containers/VideoUploadEditor/index.jsx:19]: This is required to get the close button working on the editor since this is passed as a parameter to a higher-order function.

File: src/editors/data/images/videoThumbnail.svg
Comment (xitij2000) [inline src/editors/data/images/videoThumbnail.svg:1]: I think the right place to add this icon will be in paragon. 

It is a pretty simple process that you can see [here](https://github.com/openedx/paragon/tree/master/icons#readme). You'll just need to put this svg in the correct folder, run build in the icon folder and make a PR based on the changes. 

I think you can remove this icon for now and put that in a separate PR?

File: src/editors/containers/VideoUploadEditor/index.scss [hunk 16]
```diff
@@ -4,45 +4,45 @@
...
-  border: 1px solid #707070;
-  width: 308px;
-
-  input {
-    border: none !important;
-    width: 90%;
-  }
 
   input::placeholder {
     color: #454545;
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.scss:31]: This corresponds with $gray-700, you can use that here.

File: src/editors/containers/VideoUploadEditor/index.scss [hunk 17]
```diff
@@ -4,45 +4,45 @@
...
+    // color: #5E35B1;
+    font-family: 'Inter';
+    font-weight: 500;
+    word-wrap: break-word;
```
Comment (xitij2000) [inline src/editors/containers/VideoUploadEditor/index.scss:35]: I don't think you need to specify font-family here. Inter is the font used by the [edx theme ](https://github.com/edx/brand-edx.org/blob/d4e4fa4dcd06b2fec01af97121ebd278ec95c89b/paragon/_variables.scss#L419), it is not available in the openedx theme. So if you use the edx branding variant, this font will probably already be applied, but for the the standard branding the font will be different.

**Review by xitij2000 — APPROVED** [review]
:+1:

- [x] I tested this: tested on devstack
- [x] I read through the code

**Review by kenclary — APPROVED** [review]
ok, looks good
